libs/docs/platform/smart-filter-bar/examples/platform-smart-filter-bar-custom-filter-example.component.ts

```diff
@@ -77,8 +77,13 @@ export class PlatformSmartFilterBarSliderComponent extends BaseDynamicFormGenera
     viewProviders: [dynamicFormFieldProvider, dynamicFormGroupChildProvider, smartFilterBarProvider]
 })
 export class PlatformSmartFilterBarDateRendererComponent extends BaseSmartFilterBarConditionField {
-    constructor(dialogService: DialogService, smartFilterBar: SmartFilterBar, injector: Injector) {
-        super(dialogService, smartFilterBar, injector);
+    constructor(
+        dialogService: DialogService,
+        smartFilterBar: SmartFilterBar,
+        smartFilterBarService: SmartFilterBarService,
+        injector: Injector
+    ) {
+        super(dialogService, smartFilterBar, smartFilterBarService, injector);
```

Comment:
What do you think if we use `inject` functions on the parent class, instead of the constructor params? I know it is considered as an antipattern in some frameworks, but having deps in the child class is not better imo :/

Reply:
I think it would cause unclear usage in child class. By providing dependencies explicitly, developers can undestand which providers they might want to include in the component/module. Anyway, If you could create separate ticket for such improvement with more details, I'll be glad to discuss it with you

Reply:
In platform, we have more than 2 levels of inheritance, in many places, so that's why it would make sense to use injects, it would make classes a bit less dependent on each other.  Now if you want to add something to the BaseComponent class's dependency list, you are out of luck, because many things should be changed to accommodate that small change. Cases like this are not uncommon in Platform, so that's why I think it would make sense to rely more on relatively new feature coming from Angular

Reply:
I would move this discussion to the whole team, if you don't mind.